core/controllers/base_test.py

```diff
@@ -1423,3 +1469,397 @@ def test_that_correct_derived_class_does_not_raise_exception(self):
+
+    def test_schema_keys_exactly_match_with_url_path_elements(self):
+        """This test ensures that schema keys in URL_PATH_ARGS_SCHEMAS must
+        exactly match with url path elements.
+        """
+        handlers_with_missing_url_schema_keys = []
+        list_of_routes_which_needs_schema = (
+            self._get_list_of_routes_which_need_schemas())
+
+        for route in list_of_routes_which_needs_schema:
+            handler = route.handler
+
+            handler_class_name = handler.__name__
+            if handler_class_name in self.handler_class_names_with_no_schema:
+                continue
+            if handler.URL_PATH_ARGS_SCHEMAS is None:
+                continue
+
+            regex_pattern = r'<.*?>'
+            url_path_elements = [
+                keyword[1:-1] for keyword in re.findall(
+                    regex_pattern, route.name)]
+            schema_keys = handler.URL_PATH_ARGS_SCHEMAS.keys()
+
+            missing_schema_keys = set(url_path_elements) - set(schema_keys)
+            if missing_schema_keys:
+                handlers_with_missing_url_schema_keys.append(handler_class_name)
+                self.log_line(
+                    'Missing keys in URL_PATH_ARGS_SCHEMAS for %s: %s.' % (
+                        handler_class_name, ', '.join(missing_schema_keys)))
+
+        error_msg = (
+            'Missing schema keys in URL_PATH_ARGS_SCHEMAS for [ %s ] classes.'
+            '\nVisit %s to know, how to write schema for handler args.' % (
+                ', '.join(handlers_with_missing_url_schema_keys),
+                    self.wiki_page_link))
+
+        self.assertEqual(handlers_with_missing_url_schema_keys, [], error_msg)
+
+    def test_schema_keys_exactly_match_with_request_methods_in_handlers(self):
+        """This test ensures that schema keys in HANDLER_ARGS_SCHEMAS must
+        exactly match with request arguments.
+        """
+        handlers_with_missing_request_schema_keys = []
+        list_of_routes_which_needs_schema = (
+            self._get_list_of_routes_which_need_schemas())
+
+        for route in list_of_routes_which_needs_schema:
+            handler = route.handler
+
+            handler_class_name = handler.__name__
+            if handler_class_name in self.handler_class_names_with_no_schema:
+                continue
+            if handler.HANDLER_ARGS_SCHEMAS is None:
+                continue
+
+            handler_request_methods = []
+            if handler.get != base.BaseHandler.get:
+                handler_request_methods.append('GET')
+            if handler.put != base.BaseHandler.put:
+                handler_request_methods.append('PUT')
+            if handler.post != base.BaseHandler.post:
+                handler_request_methods.append('POST')
+            if handler.delete != base.BaseHandler.delete:
+                handler_request_methods.append('DELETE')
+            schema_keys = handler.HANDLER_ARGS_SCHEMAS.keys()
```

Comment:
methods_defined_in_schema perhaps -- "keys" is a bit generic and it gives less of a picture about what those keys are.

Reply:
Done